src/main/java/jenkins/scm/api/SCMTriggerItemSCM.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016 CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package jenkins.scm.api;
+
+import hudson.scm.SCM;
+import hudson.triggers.SCMTrigger;
+import hudson.triggers.Trigger;
+import jenkins.scm.impl.SCMTriggerItemSCMNotifier;
+import jenkins.scm.impl.SCMTriggerListener;
+import jenkins.triggers.SCMTriggerItem;
+import org.apache.tools.ant.ExtensionPoint;
+
+/**
+ * Extension point to allow modern {@link SCM} implementations to auto-register call-back hooks for
+ * {@link SCMTriggerItem} instaces without having to implement a custom {@link Trigger}.
+ *
+ * @since 2.0
+ * @see SCMTriggerItemSCMNotifier
+ */
+public abstract class SCMTriggerItemSCM extends ExtensionPoint {
```

Comment:
Perhaps `SCMTriggerItemSaveListener`? At any rate, Javadoc could clarify that the purpose of this class is to allow an SCM plugin to register webhooks that will be processed via the new event system, without having to create a phony `Trigger` (one with no `run` override nor `cronTabSpec`) merely to intercept `start` (and `stop`, though currently we do not offer a replacement).

Do we offer any replacement for `getProjectActions`? For example like `GitHubWebHookPollingAction`?

Is there any downstream example of this extension point being implemented, demonstrating that it can fully replace the old idiom?